commit e5cd3551220c21afcbd1d9d94773d212e2b33800
Author: Brian Bennett <brian.bennett@joyent.com>
Date:   2019-09-04T10:21:29-07:00 (5 weeks ago)
    
    joyent/manta-thoth#95 thoth should work inside a job
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/bin/thoth b/bin/thoth
index 8dea1a6..d31ac7d 100755
--- a/bin/thoth
+++ b/bin/thoth
@@ -1,7 +1,7 @@
 #!/usr/bin/env node
 
 /*
- * Copyright 2016 Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  */
 
 var mod_ctype = require('ctype');
@@ -3549,10 +3549,12 @@ var connect = function ()
 	var i, opts = {};
 	var mvars = [ 'MANTA_URL', 'MANTA_USER', 'MANTA_KEY_ID' ];
 
-	for (i = 0; i < mvars.length; i++) {
-		if (!process.env[mvars[i]]) {
-			fatal('expected ' + mvars[i] +
-			    ' environment variable to be set');
+	if (process.env.MANTA_NO_AUTH !== 'true') {
+		for (i = 0; i < mvars.length; i++) {
+			if (!process.env[mvars[i]]) {
+				fatal('expected ' + mvars[i] +
+				    ' environment variable to be set');
+			}
 		}
 	}
 
